src/Sarif.WorkItemFiling/WorkItemFiler.cs

```diff
@@ -1,9 +1,100 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Reflection;
+using System.Text;
+using Microsoft.Json.Schema;
+using Microsoft.Json.Schema.Validation;
+
 namespace Microsoft.CodeAnalysis.Sarif.WorkItemFiling
 {
     public class WorkItemFiler
     {
+        private static readonly Validator s_logFileValidator = CreateLogFileValidator();
+        private readonly IFileSystem _fileSystem;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WorkItemFiler"> class.</see>
+        /// </summary>
+        /// <param name="fileSystem">
+        /// An object that implements the <see cref="IFileSystem"/> interface, providing
+        /// access to the file system;
+        /// </param>
+        public WorkItemFiler(IFileSystem fileSystem)
+        {
+            _fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+        }
+
+        /// <summary>
+        /// Files work items from the results in a SARIF log file.
+        /// </summary>
+        /// <param name="path">
+        /// The path to the SARIF log file.
+        /// </param>
+        public void FileWorkItems(string path)
+        {
+            if (path == null) { throw new ArgumentNullException(nameof(path)); }
+
+            ValidateLogFileAgainstSarifSchema(path);
```

Comment:
>ValidateLogFileAgainstSarifSchema [](start = 12, length = 33)

suggestion: what about a name like "EnsureIsValidSarif" etc?

Reply:
Good rename. Will do in next PR.